File: backend/src/main/java/com/codeb/ims/service/ChainService.java
package com.codeb.ims.service;

import com.codeb.ims.dto.ChainRequest;
import com.codeb.ims.dto.ChainResponse;

import java.util.List;

public interface ChainService {

    ChainResponse addChain(ChainRequest request);

    List<ChainResponse> getAllChains();

    List<ChainResponse> getChainsByGroup(Long groupId);

    ChainResponse updateChain(Long id, ChainRequest request);

    void deleteChain(Long id);
}

File: frontend/src/api/chainService.js
import api from "./axios";

const chainService = {
  getAllChains: () => api.get("/chains"),

  getChainsByGroup: (groupId) =>
    api.get(`/chains/by-group/${groupId}`),

  getChainById: (id) =>
    api.get(`/chains/${id}`),

  addChain: (data) =>
    api.post("/chains", data),

  updateChain: (id, data) =>
    api.put(`/chains/${id}`, data),

  deleteChain: (id) =>
    api.delete(`/chains/${id}`),
};

export default chainService;